### DISCUSSION TITLE
terminal output:   ERROR: [youtube] wrAAX90agNM: Sign in to confirm you’re not a bot. This helps protect our community. Learn more

### DISCUSSION THREAD

**Discussion by 111100001** [bibanon/tubeup#330] [General]
people say if you see this error it means Youtube banned my IP.
However, i know the solution to this issue.
 
by using Oauth plugin and injecting it into yt-dlp it should solve it. this is the git page: https://github.com/coletdjnz/yt-dlp-youtube-oauth2

My question is: where is the config file for yt-dlp? i dont know where to find it. i used pipx to install Tubeup.

i tested the Oauth plugin on a yt-dlp other than the one that Tubeup works with and it worked. now i just need to do it with the yt-dlp that Tubeup is working wih

**Comment by vxbinaca** [comment]
the people you read are wrong use cookies

**Comment by 111100001** [comment]
the yt-dlp contributors said never use cookies youll get cookie banned

**Comment by vxbinaca** [comment]
>My question is: where is the config file for yt-dlp? i dont know where to find it. i used pipx to install Tubeup.

It's in your venv.

**Comment by 111100001** [comment]
![Screenshot 2024-08-15 194510](https://github.com/user-attachments/assets/5073ff89-3d15-4ba3-84cd-c41b84a184eb)
![Screenshot 2024-08-15 194535](https://github.com/user-attachments/assets/613360ea-4a2d-48e0-aee5-ecaff7f7fde5)

sorry to bother, but i don't understand this. when i used yt-dlp (after adding a yt-dlp config file to bypass the "confirm you're not a bot...") to download a video, it worked fine, but when i used Tubeup, yt-dlp shows the bot error again. 

as you can see, i am working in the venv so im using the yt-dlp that is with tubeup. 

does tubeup somehow make yt-dlp not load the yt-dlp config file that i added?

**Comment by 111100001** [comment]
i guess this is similar to what i want https://github.com/bibanon/tubeup/issues/196#issue-1218463764

**Comment by vxbinaca** [comment]
no we're not having open flags it invites abuse. no I do not care about your specific use case.

Use a cookie, if it's banned get another, go slower. Use sleeps. You can either use the sleep command in Linux or use the built in sleeps function. You need to change how you behave to not appear like a scraper.

Again, open yt-dlp args will not fix this and cause a nightmare for the site. 

It's not happening.

Edit: I do not care one lick what the "top contributor" to yt-dlp says. I've pumped 300k videos into IA. The sleeps function is in yt-dlp because of me. Listen to me. Get a cookie or figure out how to edit the config in your yt-dlp instance to force oauth.

**Comment by vxbinaca** [comment]
And if you're getting the bot warning you've gotten your IP flagged, which no amount of open yt-dlp args will fix,. Wait 72 hours before ripping another video.

**Comment by vxbinaca** [comment]
also do you have yt-dlp installed elsewhere? where does your path point for yt-dlp?

edit:

Have you tried moving the config to home aka `~/`? `yt-dlp [seems to say](https://github.com/yt-dlp/yt-dlp#configuration) thats a location it will pick up the config.

**Reply by 111100001** [reply]
i do not have yt-dlp anywhere else. i was using the yt-dlp that is in the tubeup venv. 

![image](https://github.com/user-attachments/assets/4930e636-be32-462d-873f-9e6d0a7737a6)

the last time i used tubeup was almost 3 months ago. when i used it 3 days ago for the first time since, i saw that error come up.

**Reply by 111100001** [reply]
when i use yt-dlp itself that is in the tubeup venv, it WORKS just fine no problems and picks up the config file that i added.

my problem is when i use tubeup it (the yt-dlp instance that tubeup initialized) doesn't pick up the yt-dlp config. i put the config file everywhere that the yt-dlp devs say it will pick it up to no avail

**Reply by vxbinaca** [reply]
Well we don't force ignore the config so I don't know why thats happening. Check the args we use.

**Reply by vxbinaca** [reply]
@111100001 I want you to test using a cookie for just one video, and tell me if it works. I know oauth is more secure. I know yt-dlp has had cookie leaks, you should be using an account you don't care about anyway. I want you to try with a cookie and see if it works, because I pumped 1000 videos to IA with one and it worked.

**Reply by 111100001** [reply]
i will try that and get back to you

**Comment by 111100001** [comment]
using cookies only worked for the first video . the commands after that, output gave the same bot error

**Reply by vxbinaca** [reply]
Youtube is aggressively refreshing your cookies, pull newest cookie and try again. Also your IP is being blacklisted.

**Comment by 111100001** [comment]
tried to pull newer cookies, it only worked for 3 commands. 

might be too hard for me, but i'll try adding netrc args to the tubeup.py file as there is another way to authenticate using netrc

**Comment by vxbinaca** [comment]
Thats not gonna work with .netrc. Are you using sleeps?

`sleep $((RANDOM % 90)) ;  tubeup `

Go slower. Stop looking like a bot to Youtube and sucking everything down as fast as you can with one bare Tubeup command pointed at a channel. Have yt-dlp print out a channels URLs into a batch file, then make batch file with the sleeps and cookies.

`yt-dlp --get-id [CHANNEL] > batch.sh`

Use Vim to mass edit the file and have it sleep 90 seconds tops per video.

You also need to wait 72 hours at least once you start getting those IP bans before Youtube stops blacklisting you.

Edit: Are you using a VPS to rip that does nothing but rip, or are you using the same machine and IP bound to a machine you watch Youtube on? Thats another way to avoid scrape bans is to look like legitimate watching activity. 

If you read the front page, it says channel rips aren't recommended. Well we've all been living on borrowed time, and sadly not even your vaunted oauth command - and I do not care what yt-dlp does alone outside of tubeup post azll the screenshots you want - will fix this.

Go slower. Wait longer between bans at least 72 hours. Use a cookies and update them often. Rip a channel a video at a time if you have to all of these things have been working for me.

**Comment by 111100001** [comment]
i used the VPS to upload some videos back in June. i had no problems then.
after i was done i never used the VPS untill the day i contacted you, when
i discovered my ip was blacklisted (from the very first tubeup command).

so, i know waiting for 3 days is not going to work for my case (because i
stopped for 2 months before i did it again).

i do have a batch file , each line is a tubeup command , but i dont use
sleeps because it takes around an hour to upload one video to ia. how is a
90 second sleep going to help? the time the tubeup command takes to upload
its video should serve as a sleep until the next tubeup command, no?

anyways, thanks for your help

**Reply by vxbinaca** [reply]
>how is a 90 second sleep going to help?

Because theres a time between you accessing the page and a reasonable amount of time their anti-scraping algos think you watched the video. If you hammer YT for a bunch of small videos you'll get IP banned. You can't watch videos fast enough when it's a channel of small videos and yt-dlp rate bypasses and the data flies in. You need to appear more human. The large videos 

I'm not incorporating that oauth branch I want yt-dlp to merge that thing you linked and then push up a flag for it and after tests we'll import and cut a new version. I want deps to be kept to a minimum for code maintainability.

**Reply by vxbinaca** [reply]
and on that oauth plugin, remove yt-dlp from `tubeup` venv, reinstall the latest nightly in it's own venv, try to see if tubeup can see the seperate venv, then inject that oauth thing and see if that allows it to work, if it works we'll update README unless theres other code changes needed, if thats the case do a PR with a flag for oauth and we'll merge it.

**Comment by 111100001** [comment]
thanks

**Comment by 111100001** [comment]
solved. see PR
